shared/chat/conversation/info-panel/min-writer-role/index.js

```diff
@@ -0,0 +1,161 @@
+// @flow
+import * as React from 'react'
+import * as Kb from '../../../../common-adapters'
+import * as TeamTypes from '../../../../constants/types/teams'
+import * as TeamConstants from '../../../../constants/teams'
+import * as Style from '../../../../styles'
+import {upperFirst} from 'lodash-es'
+
+type Props = {
+  canSetMinWriterRole: boolean,
+  isSmallTeam: boolean,
+  minWriterRole: TeamTypes.TeamRoleType,
+  onSetNewRole: (newRole: TeamTypes.TeamRoleType) => void,
+}
+
+type State = {
+  saving: boolean,
+  selected: TeamTypes.TeamRoleType,
+}
+
+class MinWriterRole extends React.Component<Props, State> {
+  state = {saving: false, selected: this.props.minWriterRole}
+  _setSaving = (saving: boolean) => this.setState(s => (s.saving === saving ? null : {saving}))
+  _setSelected = selected => this.setState(s => (s.selected === selected ? null : {selected}))
+  _selectRole = role => {
+    if (role !== this.props.minWriterRole) {
+      this._setSaving(true)
+      this._setSelected(role)
+      this.props.onSetNewRole(role)
+    }
+  }
+
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    if (prevProps.minWriterRole !== this.props.minWriterRole) {
+      if (this.props.minWriterRole === prevState.selected) {
+        // just got value that matches ours. We aren't saving anymore
+        this._setSaving(false)
+      }
+      this._setSelected(this.props.minWriterRole)
+    }
+  }
+
+  render() {
+    const items = TeamConstants.teamRoleTypes.map(role => ({
+      onClick: () => this._selectRole(role),
+      title: upperFirst(role),
+    }))
```

Comment:
Does this need to happen inside `render()`?

Reply:
Good point - I'll move it to `componentDidUpdate`

Reply:
actually it's a little more complicated that way and I don't think this will render that often. I'll TODO and skip for now.